vms/platformvm/state/staker_test.go

```diff
@@ -133,88 +134,89 @@ func TestStakerLess(t *testing.T) {
 
 func TestNewCurrentStaker(t *testing.T) {
 	require := require.New(t)
-	ctrl := gomock.NewController(t)
+	stakerTx := generateStakerTx(require)
 
 	txID := ids.GenerateTestID()
-	nodeID := ids.GenerateTestNodeID()
-	sk, err := bls.NewSecretKey()
-	require.NoError(err)
-	publicKey := bls.PublicFromSecretKey(sk)
-	subnetID := ids.GenerateTestID()
-	weight := uint64(12345)
-	startTime := time.Now()
-	endTime := startTime.Add(time.Hour)
-	potentialReward := uint64(54321)
-	currentPriority := txs.SubnetPermissionedValidatorCurrentPriority
-
-	stakerTx := txs.NewMockStaker(ctrl)
-	stakerTx.EXPECT().EndTime().Return(endTime)
```

Comment:
🤮 

`<rant>`
Expecting a call to a test-double, such that the test fails if it doesn't occur, is too regularly a tautological test of implementation instead of behaviour. Such tests are brittle (and often useless) because changing the code requires changing the test and then there's no way to know if the new implementation is correct. Conversely, if the lack of a call doesn't result in the test failing then the method shouldn't be called `EXPECT()`.
`</rant>`